examples/datasets/load_limo_dataset.py

```diff
@@ -0,0 +1,173 @@
+    data = scipy.io.loadmat(fname_eeg)
+
+    # -- 2) get epochs information from structure
+    # sampling rate
+    sfreq = data_info['LIMO']['data'][0][0][0][0]['sampling_rate'][0][0]
+    # tmin and tmax
+    tmin = data_info['LIMO']['data'][0][0][0][0]['start'][0][0]
+    # number of epochs per condition
+    design = data_info['LIMO']['design'][0][0]['X'][0][0]
+
+    # create events matrix
+    events = np.zeros(shape=(len(design), 3), dtype=int)
+    events[:, 0] = list(range(len(design)))
+    events[:, 2] = design[:, 1]
+    # event ids, such that B == 1
+    event_id = dict(A=0, B=1)
+
+    # -- 3) extract channel labels from LIMO structure
+    # get individual labels
+    labels = data_info['LIMO']['data'][0][0][0][0]['chanlocs']['labels']
+    labels = [ll[0] for ll in labels[0]]
+    # get montage
+    montage = mne.channels.read_montage('biosemi128')
+    # add external electrodes (e.g., eogs)
+    ch_names = montage.ch_names[:-3] + ['EXG1', 'EXG2', 'EXG3', 'EXG4']
+    # match individual labels to labels in montage
+    found_inds = [ii for ii, ee in enumerate(ch_names) if ee in labels]
+    missing_chans = [ch for ch in ch_names if ch not in labels]
+    assert labels == [ch_names[ii] for ii in found_inds]
+
+    # -- 4) extract data from subjects Yr structure
+    # data is stored as channels x time points x epochs
+    # data['Yr'].shape  # <-- see here
+    # transpose to epochs x channels time points
+    obs_data = np.transpose(data['Yr'], (2, 0, 1))
+
+    # initialize data in expected order
+    all_data = np.empty((obs_data.shape[0],
+                         len(ch_names),
+                         obs_data.shape[2]))
+    # copy over the non-missing data.
+    for source, target in enumerate(found_inds):
+        # avoid copy when fancy indexing.
+        all_data[:, target, :] = obs_data[:, source, :]
+    obs_data = all_data
+    obs_data /= 1e6  # data to V (to match MNE)
+
+    # create list containing channel types
+    types = []
+    for ch in ch_names:
+        types.append(
+            'eog' if ch in ('EXG1', 'EXG2', 'EXG3', 'EXG4') else 'eeg')
+
+    # -- 5) Create custom info for mne epochs structure
+    # create info
+    info = mne.create_info(ch_names=ch_names, ch_types=types, sfreq=sfreq)
+
+    # -- 6) Create custom epochs array
+    epochs = mne.EpochsArray(obs_data, info, events, tmin, event_id)
+    epochs.set_montage(montage=montage)
+    epochs.info['bads'] = missing_chans  # missing channels are marked as bad.
+
+    # -- 7) interpolate missing channels
+    if interpolate is True and montage is not None:
+        # code bad channels
+        epochs.interpolate_bads(reset_bads=True)
+
+    return epochs, design
+
+
+# --- Use load_limo_epochs() function to fetch subject data ---
+
+# fetch data from subject 1
+data, design = load_limo_epochs(subject = 1, path=None)
+
+# inpect epochs info
+data.info
+
+# plot evoked response for condition A
+data['A'].average().plot_joint(times=[.09, .15])
```

Comment:
If the fetcher is put into a proper function, the actual example can be more interesting.

Look at the examples for e.g. the SPM data.